### DISCUSSION TITLE
ORDER BY on concatenated field taking a long time after upgrade from 24.6.2.17 to 24.9.2.42

### DISCUSSION THREAD

**Discussion by orn1983** [ClickHouse/ClickHouse#71185] [Q&A]
Hello,

I have a query on a poorly structured dataset which after upgrading ClickHouse is now taking a really long time ordering the data.
The ORDER BY clause isn't using the innate table ordering, but is rather a composite string generated by concatinating two values.

The query takes ~40 seconds when the ORDER BY is omitted, but takes  on unordered data, but adding an ORDER BY using the concatenated value takes the query time up to 310 seconds. Prior to the upgrade, the query would reliably finish in under 60 seconds.

The query finishes up to 99% in the same amount of time as for the other query, and then it looks as-if the query has just hung, with no CPU usage. The result is only 7.2 million rows. When I output the result to a file, I can monitor the process (when it looks like nothing is happening) by counting the output lines, and it seems to be ordering/writing in batches of 65409 rows at a time. Every batch takes about 2-3 seconds.

Is this expected behavior? I found some older threads from 2019 where people were complaining about similar issues but didn't find anything recent.

optimize_read_in_order is 1 and I have no memory limitations on max_bytes_before_external_sort or max_memory_usage.

**Comment by filimonov** [comment]
Hard to say something concrete w/o looking on the query / schema. 

If it's still the case - show the query, explain, schema, the query finish row from system.query_log, and run that query with `send_logs_level='trace'`  in clickhouse-client. Then share / attach all the details.
